### PR TITLE
Bugs 1695622,1699129: bump buildah to v1.8.2, pin most dependencies

### PR DESCRIPTION
**Pull request by nalind**
Update the version of buildah that we use to version 1.8.0, but downgrade the version of storage that we use to an older version.

### CODE REVIEW

**Comment by nalind** [comment]
It looks like we broke something for builds that use post-commit hooks.

**Comment by nalind** [comment]
Updated to be a bit newer than 1.8.0, closer to what I expect 1.8.1 to be, and pinning most other dependencies.

**Comment by nalind** [comment]
Bumped to the commit ID that will be 1.8.1.

**Comment by nalind** [comment]
Moved our buildah reference in glide.yaml to the v1.8.1 tag instead of the commit ID that it points to.  Pulled in some changes to the openshift branch of kubernetes.

**Comment by nalind** [comment]
Bumped to 1.8.2.

File: pkg/build/builder/common.go [hunk 1]
```diff
@@ -14,8 +14,8 @@ import (
 	"time"
 
 	"github.com/docker/distribution/reference"
-	dockercmd "github.com/docker/docker/builder/dockerfile/command"
-	"github.com/docker/docker/builder/dockerfile/parser"
+	dockercmd "github.com/openshift/imagebuilder/dockerfile/command"
+	"github.com/openshift/imagebuilder/dockerfile/parser"
```
Comment (adambkaplan) [inline pkg/build/builder/common.go]: Why the change from docker/docker/builder to imagebuilder?

Comment (nalind) [reply pkg/build/builder/common.go]: imagebuilder incorporated it to avoid versioning conflicts, so the type we get from `imagebuilder.ParseDockerfile()` no longer matched the argument type for local functions like `nodeHasFromRef()`.

**Comment by adambkaplan** [comment]
/lgtm

Unclear why image-ecosystem tests flaked twice on the perl/dancer image.
